### ISSUE TITLE
Implement category blacklist / whitelist

### ISSUE THREAD

**Issue by PuppyLover101** [cross-seed/cross-seed#395]
Would be very useful if we could add a blacklist / whitelist to categories, so certain categories aren't cross-seeded.

Personally, I use the post-import feature on sonarr to move my torrents to another category with another save path after importing to sonarr. Cross-seeding always fails for torrents in this stage, as there are no files on tv-sonarr.crosseed save path.

Thank you.

**Comment by mokahless** [comment]
I'd also love to see a feature like this, but with tags. I have autobrr, sonarr and radarr going and a lot of content from there is set to autodelete after x weeks. If cross-seed runs, it will add permanent versions of these temporary torrents, for the other trackers. Right now, I have to manually go in every once in a while to clear out the torrents cross-seed added, once sonarr/radarr/autobrr deletes the original torrent.

So blacklisting anything with either the relevant sonarr/radarr category or the autobrr tag would be very useful for me.

**Comment by robpaolella** [comment]
Strongly agree on getting this added. Would make a huge difference for categories I have like seed-limit, which I have just to satisfy seed limits that weren't met for one reason or another.

**Comment by onedr0p** [comment]
I have a `hit-and-run` category that I manually add torrents to if my indexer give me one (sometimes it just happens), it would be nice to omit that category from cross-seed.

**Comment by dylancmds** [comment]
Would love this as well. I use autobrr to snatch and throw away torrents on trackers without HnR rules, cross-seed picks these up and injects them from other trackers. Any way to configure an ignored tag or category can help remedy this. I'd hate to stand up a separate client just for autobrr.

**Comment by AndresPineros** [comment]
Yup, this would be great! I have like 1000 book torrents that I don't want to cross-seed because books are in the kilobytes, so there's no point cross-seeding those. Plus, I don't want to abuse the indexers with these unnecessary requests, or make my cross-seed scan super slow to prevent abusing the APIs.

It can be solved in the mean time with the hooks offered by the torrent clients. In qbittorrent it is the " Run external program on torrent finished". Just validate if the category/tags match the ones you want and hardlink/copy of the torrent file to a custom folder. Make cross-seed read from that folder instead of the suggested ones.

**Comment by ShanaryS** [comment]
If this is to be implemented, it should be per indexer similar to #612. This gives the added option of ignoring movie searches for tv trackers and vice versa. Would be nice if dataDirs could be used additionally as well.

**Comment by zakkarry** [comment]
> If this is to be implemented, it should be per indexer similar to #612. This gives the added option of ignoring movie searches for tv trackers and vice versa. Would be nice if dataDirs could be used additionally as well.

This functionality is in the pipeline (completed and being tested further for v6) as you know - as far as category blacklisting goes....this would basically need to be done during prefiltering most likely, and would require #471 if we were to implement it properly (otherwise it would be post-snatch and during injection that it aborts - and that's not really accomplishing the grander scope of the functionality to reduce searches....although technically still blocking the release.

Basically, we would have to implement this in Deluge with labels as well as qBittorrent, and that would require expanding the API usages and data structuring in our current codebase...

While this seems like a simple feature, the proper implementation requires quite a bit of effort and dedication to get functional so we wouldn't later be rewriting the entire thing for sqlite (whenever that happens)...

**Comment by zakkarry** [comment]
Given the seeming demand for this feature, I've joined forces with @bakerboy448 who maintains some third-party scripts, and written up a category and tracker blacklist/whitelist bash script you can use to execute the on complete cross-seed command.

It's a basic bash script, and is fairly simple to configure. Simple google search will be able to help you add the necessary commands if you do not understand.

I understand this is not integrated into cross-seed natively, but I figured something was better than nothing for you guys.

Change the `$TORRENT_CAT` at the bottom to `$TORRENT_TRACKER` to match the regex on the right side of the `=~`

https://github.com/bakerboy448/StarrScripts/blob/main/xseed_qbit_cat_filter.sh

**Comment by zakkarry** [comment]
> I thought of a similar feature request, but instead of allow/deny list, it would allow the ability for groups to be created which cross-seed with each other. So you may have A/B/C categories that search across one another, and X/Y/Z that are independently searched, which reduces noise from low value searches.

While this could be a benefit if you happen to know one specific tracker cross-seeds well, the overall use-case for this seems rather niche. 

We've already implemented measures into the v6 pre-release that prevent almost all of the unnecessary searches (content type wise) from hitting trackers that do not support that media.

https://www.cross-seed.org/docs/v6-migration#torznab-categories

What you're saying could cut down on the low-value searches, but there's really no harm in searching and effectively knowing what content is on another tracker already beyond a media type (TV, movies, anime, etc) is probably not that viable of a plan.

**Comment by ShanaryS** [comment]
Implemented in #773
